x-pack/plugins/stack_connectors/server/connector_types/lib/servicenow/service.ts

```diff
@@ -249,6 +255,81 @@ export const createExternalService: ServiceFactory = ({
     }
   };
 
+  const getIncidentByCorrelationId = async (correlationId: string): Promise<ServiceNowIncident> => {
+    try {
+      const res = await request({
+        axios: axiosInstance,
+        url: getIncidentByCorrelationIdUrl(correlationId),
+        method: 'get',
+        logger,
+        configurationUtilities,
+      });
+
+      checkInstance(res);
+
+      const foundIncident = useTableApi ? res.data.result : res.data.result[0];
```

Comment:
I don't think we need this check here. We use the Table API (`tableApiIncidentUrl`) for this request. The results will always be an array of items. Example:

```
{
    "result": [
        {
            "sys_id": "12231",
            "short_description": "test itsm",
            "sys_created_on": "2023-10-27 13:23:46",
            "correlation_id": "5869"
        }
    ]
}
```